cub/cub/thread/thread_operators.cuh

```diff
@@ -414,121 +467,283 @@ _CCCL_HOST_DEVICE BinaryFlip<BinaryOpT> MakeBinaryFlip(BinaryOpT binary_op)
   return BinaryFlip<BinaryOpT>(binary_op);
 }
 
+/***********************************************************************************************************************
+ * SIMD Operators
+ **********************************************************************************************************************/
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
+
+// TODO: extend to floating_point<M, E>
+
 namespace internal
 {
-// TODO: Remove DPX specilization when nvbug 4823237 is fixed
 
 template <typename T>
-struct DpxMin
+struct SimdMin
 {
-  static_assert(detail::always_false<T>(), "DpxMin is not supported for this type");
+  static_assert(cub::detail::always_false<T>(), "Unsupported specialization");
 };
 
 template <>
-struct DpxMin<::cuda::std::int16_t>
+struct SimdMin<::cuda::std::int16_t>
 {
+  using simd_type = unsigned;
+
   _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE unsigned operator()(unsigned a, unsigned b) const
   {
     return __vmins2(a, b);
   }
 };
 
 template <>
-struct DpxMin<::cuda::std::uint16_t>
+struct SimdMin<::cuda::std::uint16_t>
 {
+  using simd_type = unsigned;
+
   _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE unsigned operator()(unsigned a, unsigned b) const
   {
     return __vminu2(a, b);
   }
 };
 
+#  if defined(_CCCL_HAS_NVFP16)
+
+template <>
+struct SimdMin<__half> : cub::Min
+{
+  using simd_type = __half2;
+
+  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __half2 operator()(__half2 a, __half2 b) const
+  {
+    NV_IF_TARGET(NV_PROVIDES_SM_80,
+                 (return __hmin2(a, b);),
+                 (return __half2{static_cast<__half>(cub::Min{}(static_cast<float>(a.x), static_cast<float>(b.x))),
+                                 static_cast<__half>(cub::Min{}(static_cast<float>(a.y), static_cast<float>(b.y)))};));
```

Comment:
This is not valid if the user disables conversions / constructors
```suggestion
                 (return __halves2half2{__float2half(cub::Min{}(__half2float(a.x), __half2float(b.x))),
                                        __float2half(cub::Min{}(__half2float(a.y), __half2float(b.y)))};));
```

Applies throughout